File: src/main/java/banane/io/pdb/model/MapPoint.java
package banane.io.pdb.model;

import javax.persistence.*;

@Entity
public class MapPoint {

    private Long id;

    private int x;

    private int y;

    private String zone;

    @Enumerated(EnumType.STRING)
    private Terrain terrain;


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public String getZone() {
        return zone;
    }

    public void setZone(String zone) {
        this.zone = zone;
    }

    public Terrain getTerrain() {
        return terrain;
    }

    public void setTerrain(Terrain terrain) {
        this.terrain = terrain;
    }
}

File: src/main/java/banane/io/pdb/model/MapPoint.kt
package banane.io.pdb.model

import javax.persistence.*

@Entity
class MapPoint {
    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
    @get:Id
    var id: Long? = null
    var x = 0
    var y = 0
    var zone: String? = null
    @Enumerated(EnumType.STRING)
    var terrain: Terrain? = null

}